composablejs/packages/bootstrap_pallets/src/bootstrap/crowdloanRewards/index.ts

```diff
@@ -1,58 +1,74 @@
+): [PalletCrowdloanRewardsModelsRemoteAccount, u128, u32] {
+  if (account.startsWith("0x")) {
+    return [
+      api.createType("PalletCrowdloanRewardsModelsRemoteAccount", {
+        Ethereum: account
+      }),
+      api.createType("u128", toChainUnits(reward).toString()),
+      api.createType("u32", vestingPeriod)
+    ] as [PalletCrowdloanRewardsModelsRemoteAccount, u128, u32];
+  } else {
+    return [
+      api.createType("PalletCrowdloanRewardsModelsRemoteAccount", {
+        RelayChain: api.createType("AccountId32", account).toU8a()
+      }),
+      api.createType("u128", toChainUnits(reward).toString()),
+      api.createType("u32", vestingPeriod)
+    ] as [PalletCrowdloanRewardsModelsRemoteAccount, u128, u32];
+  }
+}
 
-  const txCalls = [];
-  const allAccounts = config.crowdloanRewards.ethereumContributors.concat(
-    config.crowdloanRewards.relayChainContributors
-  );
+export async function bootstrapCrowdloanRewards(api: ApiPromise, walletSudo: KeyringPair): Promise<void> {
+  const allRewards = Object.entries(rewards);
 
-  const rewardsPerAccount = api.createType("u128", config.crowdloanRewards.rewardsPerAccount);
-  const vestingPeriod = api.createType("u32", config.crowdloanRewards.vestingPeriod);
+  const STEP = 5;
+  for (let i = 0; i < allRewards.length; i += STEP) {
+    let accountsOfBatch: [PalletCrowdloanRewardsModelsRemoteAccount, u128, u32][] = [];
+    
+    let accIndex = i;
+    let amount = new BigNumber(0);
+    while (accIndex < allRewards.length && accIndex < STEP + i) {
+      amount = amount.plus(allRewards[accIndex][1]);
+      accountsOfBatch.push(toPalletCrowdloanRewardsModelsRemoteAccount(
+        api,
+        allRewards[accIndex][0],
+        allRewards[accIndex][1],
+        config.crowdloanRewards.vestingPeriod
+      ))
+      accIndex = accIndex + 1;
+    }
 
-  while (index < allAccounts.length) {
-    const accounts = allAccounts.slice(index, index + step > allAccounts.length ? allAccounts.length : step);
-    const batchAccounts = accounts.map(account => {
-      if (account.startsWith("0x")) {
-        return [
-          api.createType("PalletCrowdloanRewardsModelsRemoteAccount", {
-            Ethereum: account
-          }),
-          rewardsPerAccount,
-          vestingPeriod
-        ] as [PalletCrowdloanRewardsModelsRemoteAccount, u128, u32];
-      } else {
-        [
-          api.createType("PalletCrowdloanRewardsModelsRemoteAccount", {
-            RelayChain: api.createType("AccountId32", account).toU8a()
-          }),
-          rewardsPerAccount,
-          vestingPeriod
-        ] as [PalletCrowdloanRewardsModelsRemoteAccount, u128, u32];
-      }
-    }) as [PalletCrowdloanRewardsModelsRemoteAccount, u128, u32][];
+    logger.info(`Adding Funds to Crowdloan: ${amount.toString()}`);
+    await addFundsToCrowdloan(
+      api,
+      walletSudo,
+      api.createType("u128", toChainUnits(amount).toString()),
+      config.crowdloanRewards.palletAccountId
+    );
```

Comment:
NOTE:
It should no longer be necessary to do this before calling `populate`. This still works fine, just an FYI.

Reply:
@PoisonPhang i'm keeping it this way for now, as it can be done both ways